core/__tests__/models/source/source.ts

```diff
@@ -740,6 +740,12 @@ describe("models/source", () => {
 
       await source.setMapping({ userId: "userId" });
     });
+
+    test("will throw error when mapping to own property and primary key is owned by other source", async () => {
+      await expect(source.setMapping({ email: "myEmail" })).rejects.toThrow(
+        /cannot map 'email' to own Property 'myEmail'/
+      );
+    });
```

Comment:
This is the new test - the other changes are cleanup / clarifications based on this new behavior.  Initially I would have guessed this change would have required a more complex series of tests... but... it doesn't! We've got other tests confirming that the first property (bootstrap) still works like expect